osu.Server.Queues.ScoreStatisticsProcessor/Processors/MedalAwarders/StarRatingMedalAwarder.cs

```diff
@@ -0,0 +1,179 @@
+﻿using JetBrains.Annotations;
+using MySqlConnector;
+using osu.Game.Beatmaps;
+using osu.Game.Online.API.Requests.Responses;
+using osu.Game.Rulesets;
+using osu.Game.Rulesets.Difficulty;
+using osu.Game.Rulesets.Mania.Mods;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Scoring;
+using osu.Server.Queues.ScoreStatisticsProcessor.Helpers;
+using osu.Server.Queues.ScoreStatisticsProcessor.Models;
+using osu.Server.Queues.ScoreStatisticsProcessor.Stores;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Beatmap = osu.Server.Queues.ScoreStatisticsProcessor.Models.Beatmap;
+
+namespace osu.Server.Queues.ScoreStatisticsProcessor.Processors.MedalAwarders
+{
+    [UsedImplicitly]
+    public class StarRatingMedalAwarder : IMedalAwarder
+    {
+        private BeatmapStore? beatmapStore;
+
+        public bool RunOnFailedScores => false;
+
+        public IEnumerable<Medal> Check(SoloScoreInfo score, UserStats userStats, IEnumerable<Medal> medals, MySqlConnection conn, MySqlTransaction transaction)
+        {
+            return checkAsync(score, medals, conn, transaction).Result;
+        }
+
+        private async Task<IEnumerable<Medal>> checkAsync(SoloScoreInfo score, IEnumerable<Medal> medals, MySqlConnection conn, MySqlTransaction transaction)
+        {
+            List<Medal> earnedMedals = new List<Medal>();
+
+            Ruleset ruleset = LegacyRulesetHelper.GetRulesetFromLegacyId(score.RulesetID);
+            Mod[] mods = score.Mods.Select(m => m.ToMod(ruleset)).ToArray();
+
+            // Ensure the score isn't using any difficulty reducing mods
+            if (mods.Any(MedalHelpers.IsDifficultyReductionMod))
+                return Enumerable.Empty<Medal>();
+
+            // On mania, these medals cannot be triggered with key mods and Dual Stages
+            if (score.RulesetID == 3 && mods.Any(isManiaDisallowedMod))
+                return Enumerable.Empty<Medal>();
+
+            try
+            {
+                beatmapStore ??= await BeatmapStore.CreateAsync(conn, transaction);
```

Comment:
Not sure about creating a second store here, rather than sharing the existing one (see https://github.com/ppy/osu-queue-score-statistics/blob/9f99f91f31087ff70d060d27ebd120f9b78f2195/osu.Server.Queues.ScoreStatisticsProcessor/Processors/ScorePerformanceProcessor.cs#L118). I'll probably look to combine these usages.

Reply:
Funny you should mention that, I did that in https://github.com/ppy/osu-queue-score-statistics/pull/230, so for sure.